docs/DeveloperGuide.adoc

```diff
@@ -844,14 +901,102 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+== Use case: Undo action
+
+*MSS*
+
+1.  User requests to undo the previous action taken.
+2.  AddressBook undo the the action.
++
+Use case ends.
+
+
+*Extensions*
+[none]
+* 2a. If there is no previous action taken.
++
+Use case ends.
+
+[discrete]
+== Use case: Find person
+
+*MSS*
+
+1.  User requests to find person and enters any keywords in the name of the person.
+2.  AddressBook returns the person requested.
++
+Use case ends.
+
+
+*Extensions*
+[none]
+* 2a.  The list of people is empty.
++
+[none]
+** 2a1.  AddressBook shows an error message.
++
+Use case ends.
+
+* 2b.  The person is not in the list.
++
+** 2b1.  AddressBook returns error message.
++
+Use case ends.
+
+[discrete]
+== Use case: Delete old contacts
+
+*MSS*
+
+1.  User requests to delete old contacts.
+2.  AddressBook prompts user for cut off date input.
+3.  User enters cut off date.
+4.  AddressBook prompts user for confirmation.
+5.  User confirms deletion.
+6.  AddressBook deletes all contact created before the cut off date.
+7.  AddressBook shows success message.
++
+Use case ends.
+
+
+*Extensions*
+[none]
+* 3a.  If user input invalid date.
++
+[none]
+** 3a1.  AddressBook prompts user for cut off date input.
+** 3a2.  User enters cut off date.
++
+Steps 3a1 - 3a2 are repeated until date entered is correct.
++
+Use case resumes from step 4.
+
+* 5a.  If user decline the deletion.
++
+Use case ends.
+
+* 6a.  If there is no contact before the cut off date
+** 6a1.  AddressBook shows error message
++
+Use case ends.
+
+
 _{More to be added}_
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `1.8.0_60` or higher installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 5000 persons without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+. Must be aesthetically appealing
```

Comment:
Quite vague. How about saying something like, UI follows Googles Material Design principles which is more objective